### DISCUSSION TITLE
Heiman lamp & remote

### DISCUSSION THREAD

**Discussion by MisterAG** [Koenkk/zigbee2mqtt#9369] [Devices]
Hi,

Noob here: I've setup zigbee2mqtt and Home assistant as docker images on a RPI4.

I ordered a couple of these Heiman light bulbs and remotes (13€ when I bought them):
https://www.amazon.de/dp/B096M2BDSR/ref=pe_27091401_487027711_TE_SCE_dp_1

I got the bulb to work by adding the following config based on a generic light bulb, it identifies as "Temperlight":
```
const definition = {
    zigbeeModel: ['TemperLight'],
    model: 'Heiman123',
    vendor: 'Heiman',
    description: 'Heimanlamp',
    extend: extend.light_onoff_brightness_colortemp(),
};
```
This works although the colortemp sliders maxes out at 370 I believe. Not sure how to adapt that.

The remote from the outside looks identical like this other Heiman HS2SS remote, with the same number of buttons:
https://www.zigbee2mqtt.io/devices/HS2SS.html

I did a quick hack of the heiman.js file (I know I will lose the config if I redeploy my docker container), but don't really now how to proceed. I changed the modelID to what I saw in the logs, and changed the 'exposes': entry.
```
{                                                                                      
        fingerprint: [{modelID: 'TempDimmerSw-EM-3.0', manufacturerName: 'HEIMAN'}],
        model: 'HS2SS',                             
        vendor: 'HEIMAN',
        description: 'Smart scene switch',                       
        fromZigbee: [fz.battery, fz.heiman_scenes],
        exposes: [e.battery(), e.action(['arrow_left_click', 'arrow_right_click', 'brightness_down_click', 'brightness_up_click', 'toggle'])],
        toZigbee: [],                              
        configure: async (device, coordinatorEndpoint, logger) => {              
            const endpoint = device.getEndpoint(1);              
            await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg', 'heimanSpecificScenes']);
            await reporting.batteryPercentageRemaining(endpoint, {min: constants.repInterval.MINUTES_5, max: constants.repInterval.HOUR});
        }, 
```  
I was able to pair the device but other than that it doesn't work. I also tried this as exposes entry:
```
exposes: [e.battery(), e.action(['arrow_left_click', 'arrow_left_hold', 'arrow_left_release', 'arrow_right_click',    
            'arrow_right_hold', 'arrow_right_release', 'brightness_down_click', 'brightness_down_hold', 'brightness_down_release', 'brightness_up_click', 'brightness_up_hold', 'brightness_up_release', 'toggle'])],
```

In the logs I only see this now and then, but not every time I click a button, logging is set to debug:
` MQTT publish: topic 'zigbee2mqtt/MyHeimanRemote', payload '{"action":null,"battery":100,"linkquality":33}'`

In home assistant I can see every timestamp with every click I do on the remote, when I hover the action 'none' marked in blue. 

How can I investigate this further? Most probably I going about it all wrong...

Best regards,
Jan

**Comment by nunocm** [comment]
I have a similar HEIMAN device (the RGB one) and made the remote work with the following:
```javascript
{
        fingerprint: [{ modelID: 'ColorDimmerSw-EM-3.0', manufacturerName: 'HEIMAN' }],
        model: 'heiman-color-dimmer',
        vendor: 'HEIMAN',
        description: 'Remote dimmer and color control',
        fromZigbee: [fz.battery, fz.command_on, fz.command_off, fz.command_move, fz.command_stop, fz.command_move_to_color],
        exposes: [e.battery(), e.action(['on', 'off', 'move', 'stop', 'color_move'])],
        toZigbee: [],
        configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint = device.getEndpoint(1);
            await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg', 'genOnOff', 'genLevelCtrl', 'lightingColorCtrl']);
            await reporting.batteryPercentageRemaining(endpoint, { min: constants.repInterval.MINUTES_5, max: constants.repInterval.HOUR });
        },
    },
```
The brightness buttons are sending messages like `{"action":"brightness_move_down","action_rate":50,"battery":41,"linkquality":58}` (followed by `brightness_stop` when releasing) and the arrow buttons send `{"action":"color_move","action_color":{"x":0.402,"y":0.573},"action_transition_time":10,"battery":41,"linkquality":58}`. Since the later cycles through a predefined set of colors, I check the position in an array to use it as "scenes" (`x` values: `[0.443,  0.402,  0.363,  0.304,  0.255,  0.234,  0.222,  0.208,  0.189,  0.228,  0.296,  0.344,  0.389,  0.47,  0.666,  0.333,  0.53]`).

Like you it was a quick hack though trial and error and probably this is not the best way to do it (hence why I didn't submit a PR), but hope it helps.

**Reply by Koenkk** [reply]
>  this is not the best way to do it (hence why I didn't submit a PR), but hope it helps.

This looks good actually, if all buttons emit an action I suggest making a PR

**Comment by nunocm** [comment]
Here it is: https://github.com/Koenkk/zigbee-herdsman-converters/pull/4660

Now I should open a PR on this repository adding this device documentation correct?

**Reply by Koenkk** [reply]
> Now I should open a PR on this repository adding this device documentation correct?
 
Yes, preferably with a picture (150x150 jpg)
